pkg/eval/json_evaluator_model.go

```diff
@@ -85,9 +85,23 @@ func (f Flags) Update(logger *logger.Logger, source string, ff Flags) map[string
 }
 
 // Delete matching flags from source. The implementation is not thread safe
+// If ff is nil, or, ff.Flags is of length 0, all flags from the given source are deleted
 func (f Flags) Delete(logger *logger.Logger, source string, ff Flags) map[string]interface{} {
 	notifications := map[string]interface{}{}
 
+	if ff.Flags == nil || len(ff.Flags) == 0 {
```

Comment:
```suggestion
	if len(ff.Flags) == 0 {
```
if nil, len is also 0